chat-android/src/main/java/com/ably/chat/Utils.kt

```diff
@@ -217,3 +226,60 @@ internal class LatestJobExecutor {
         }
     }
 }
+
+/**
+ * A custom implementation using Channel that supports optional awaitable emissions.
+ * This class allows emitting values with optional completion tracking.
+ */
+internal class AwaitableChannel<T>(private val logger: Logger) {
+    private val channel = Channel<Pair<T, CompletableDeferred<Unit>>>(Channel.UNLIMITED)
+    private val completionDeferred = AtomicReference<CompletableDeferred<Unit>?>(null)
+    private var activeCollector = false
```

Comment:
if we want to check that collect run once, it's better to use:

```kotlin
private val activeCollector = AtomicBoolean(false)

if (!activeCollector.compareAndSet(false, true)) sdkError()
```

Reply:
Yeah, actually coderabbitai suggested this. I was looking at use-case, since we call active collector only once .. i thought we should be good with normal Boolean. 
But I will check to update it

Reply:
it's just if we are building it thread-safe, it's better to make all methods thread-safe

Reply:
https://github.com/ably/ably-chat-kotlin/pull/124/commits/6d7ba0873a127246377a4423ef7fcb5729bcb167